client/public_endpoints.go

```diff
@@ -0,0 +1,237 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+// ---------------------------- Unprotected Endpoints ----------------------------
+
+func (c *Client) Health(ctx context.Context) (HealthResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.healthURL.String(), nil)
+	if err != nil {
+		return HealthResponse{}, err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return HealthResponse{}, err
+	}
+	defer resp.Body.Close()
+
+	var healthResponse HealthResponse
+	err = json.NewDecoder(resp.Body).Decode(&healthResponse)
+	if err != nil {
+		return HealthResponse{}, err
+	}
+
+	return healthResponse, nil
+}
+
+/*
+Track 24 hour rolling window price changes to receive stats about a certain market.
+For example 24 hour low, high, average price, price change and more.
+*/
+func (c *Client) Ticker(ctx context.Context, market string) (TickerResponse, error) {
+	if len(market) == 0 {
+		return TickerResponse{}, ErrMarketNotProvided
+	}
+
+	tickerURL := *c.tickerURL
+	params := tickerURL.Query()
+	params.Set("market", market)
+	tickerURL.RawQuery = params.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, tickerURL.String(), nil)
+	if err != nil {
+		return TickerResponse{}, err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return TickerResponse{}, err
+	}
+	defer resp.Body.Close()
+
+	var tickerResponse TickerResponse
+	err = json.NewDecoder(resp.Body).Decode(&tickerResponse)
+	if err != nil {
+		return TickerResponse{}, err
+	}
+
+	return tickerResponse, nil
+}
+
+// Track 24 hour rolling window price changes to receive stats about all markets.
+func (c *Client) AllTickers(ctx context.Context) (AllTickerResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.tickerURL.String(), nil)
+	if err != nil {
+		return AllTickerResponse{}, err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return AllTickerResponse{}, err
+	}
+	defer resp.Body.Close()
+
+	var allTickerResponse AllTickerResponse
+	err = json.NewDecoder(resp.Body).Decode(&allTickerResponse)
+	if err != nil {
+		return AllTickerResponse{}, err
+	}
+
+	return allTickerResponse, nil
+}
+
+// TODO make it simpler
+// candle stick
+func (c *Client) CandleStick(ctx context.Context, market string, options CandleStickOptions) (CandleStickResponse, error) {
+	if len(market) == 0 {
+		return CandleStickResponse{}, ErrMarketNotProvided
+	}
+
+	candleStickURL := *c.candleStickURL
+	params := candleStickURL.Query()
+
+	params.Set("market", market)
+
+	if options.Limit != 0 {
+		params.Set("limit", fmt.Sprintf("%d", options.Limit))
+	} else {
+		params.Set("limit", fmt.Sprintf("%d", CANDLESTICK_DEFAULT_LIMIT))
```

Comment:
these are optional params, so skip them if user doesn't provide them. This will make sure the package is in sync if we change the default values of the api

Reply:
So you saying I should not put the default values ?

Reply:
its possible that say we change the max acceptable value for limit in backend to 10 and your default is 15, that would break the package untill we update it. So its better to not provide default values and let the backed pick default value.

Reply:
please look into this

Reply:
Fixed this